testng-core-api/src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -296,4 +297,12 @@ default Class<?>[] getParameterTypes() {
   default boolean isIgnoreFailure() {
     return false;
   }
+
+  /**
+   * @return - A <code>{@link UUID}</code> that represents a unique id which is associated with
+   *     every test class object.
+   */
+  default UUID getInstanceId() {
```

Comment:
It should be removed as the class doesn't extend an interface that requires it. The instanceof will be done on that interface and only methods that implement both interfaces will be included.

Reply:
No. This cannot be removed. Irrespective of the type of instance associated with `ITestNGMethod` the id needs to be available on the interface. My first iteration on this implementation was as you had suggested. The `instanceof` check is a nightmare and there's really no clear control around where it should be applied and where it shouldn't be. That is why I decided that the instance id should be available on the API irrespective of the type of instance.

Reply:
@juherr - I decided to take your word but follow a different approach than my earlier one (wherein I was doing `instanceof` checks on the Object). Now I do the same thing but on the `ITestNGMethod` instance. I have pushed my changes as an additional commit so that it's easy for you to see the change. Please check now.